src/ripple/rpc/tests/KeyGeneration.test.cpp

```diff
@@ -164,10 +197,50 @@ class KeypairForSignature_test : public ripple::TestSuite
         testLegacySecret (secp256k1_strings.master_seed_hex);
     }
 
+    void testInvalidKeyType (char const* keyType)
+    {
+        testcase (keyType);
+
+        Json::Value params;
+        params[jss::key_type] = keyType;
+        params[jss::passphrase] = common::passphrase;
+
+        Json::Value error;
+        (void) keypairForSignature (params, error);
```

Comment:
Well, it populates `error`.  That's not a function result, but as an output parameter it's not strictly a side effect either.  (Like any idiom, you have to know it in order to understand the message.)